### PR TITLE
Add DNS Hub and Spoke

### PR DESCRIPTION
**Pull request by daniel-cit**
Creation of:

- DNS Hub project 
- DNS Hub VPC with two subnetworks
- DNS policy
- DNS Forwarding zone in DNS Hub project
- DNS Peering zone in each shared vpc project
- 4 cloud routers in DNS Hub project
- VLAN attachments for dedicated interconnect
- HA VPN as alternative communication to onprem

Both Interconnect and HA VPN section are commented. The user will need to choose which one to use.

@bharathkkb @morgante @rjerrems  could you please take a look?

### CODE REVIEW

**Comment by daniel-cit** [comment]
> Thanks for the pr @daniel-cit - I think we should separate this out.
> 
> * Create the DNS hub project in 1-org
> * Create all the other resources in the 3-networks in an environment called `shared` (cuts across all environments)
> 
> This will allow you to re-use existing modules (interconnect & VPN) and maintain a cleaner separation for what is ultimately networking code.

Done :)

File: 1-org/terraform.example.tfvars [hunk 1]
```diff
@@ -30,6 +30,14 @@ default_region = "australia-southeast1"
 
 scc_notification_name = ""
 
+dns_default_region1 = "us-central1"
```
Comment (rjerrems) [inline 1-org/terraform.example.tfvars:33]: Can we remove these from org TF vars now?

File: 3-networks/envs/shared/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# 3-networks/dev
```
Comment (rjerrems) [inline 3-networks/envs/shared/README.md:1]: should this be networks/dev or shared?

File: 3-networks/envs/shared/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+The purpose of this step is to setup private and restricted shared VPCs with default DNS, NAT (optional), Private Service networking, VPC service controls, onprem dedicated interconnect, onprem VPN and baseline firewall rules for environment dev.
```
Comment (rjerrems) [inline 3-networks/envs/shared/README.md:3]: This is not quite accurate for this env / the code it is executing

File: 1-org/terraform.example.tfvars (hunk 1 above)
Comment (daniel-cit) [reply 1-org/terraform.example.tfvars]: removed

File: 3-networks/envs/shared/README.md (hunk 3 above)
Comment (daniel-cit) [reply 3-networks/envs/shared/README.md]: fixed

File: 3-networks/envs/shared/README.md (hunk 2 above)
Comment (daniel-cit) [reply 3-networks/envs/shared/README.md]: should be `shared`. Fixed.

**Pull request merged**